runtime/service_integ_test.go

```diff
@@ -533,3 +533,98 @@ vdf  254:80   0  512B  0 disk`
 			"context cancelled while waiting for container %s to exit, err: %v", containerName, ctx.Err())
 	}
 }
+
+func startAndWaitTask(ctx context.Context, t *testing.T, c containerd.Container) string {
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+
+	task, err := c.NewTask(ctx, cio.NewCreator(cio.WithStreams(nil, &stdout, &stderr)))
+	require.NoError(t, err, "failed to create task for container %s", c.ID())
+
+	exitCh, err := task.Wait(ctx)
+	require.NoError(t, err, "failed to wait on task for container %s", c.ID())
+
+	err = task.Start(ctx)
+	require.NoError(t, err, "failed to start task for container %s", c.ID())
+
+	select {
+	case exitStatus := <-exitCh:
+		require.NoError(t, exitStatus.Error(), "failed to retrieve exitStatus")
+		require.Equal(t, uint32(0), exitStatus.ExitCode())
+		require.Equal(t, "", stderr.String())
+	case <-ctx.Done():
+		require.Fail(t, "context cancelled",
+			"context cancelled while waiting for container %s to exit, err: %v", c.ID(), ctx.Err())
+	}
+
+	return stdout.String()
+}
+
+func TestRemoveDir_Isolated(t *testing.T) {
+	internal.RequiresIsolation(t)
+	const vmID = 0
+
+	ctx := namespaces.WithNamespace(context.Background(), "default")
+
+	client, err := containerd.New(containerdSockPath, containerd.WithDefaultRuntime(firecrackerRuntime))
+	require.NoError(t, err, "unable to create client to containerd service at %s, is containerd running?", containerdSockPath)
+	defer client.Close()
+
+	image, err := client.Pull(ctx, debianDockerImage, containerd.WithPullUnpack, containerd.WithPullSnapshotter(naiveSnapshotterName))
+	require.NoError(t, err, "failed to pull image %s, is the the %s snapshotter running?", debianDockerImage, naiveSnapshotterName)
+
+	// TODO once Noah's immutable rootfs change is merged, we can use that as our rootfs for all VMs instead of copying
+	// one per-VM
+	rootfsBytes, err := ioutil.ReadFile(defaultVMRootfsPath)
+	require.NoError(t, err, "failed to read rootfs file")
+	rootfsPath := fmt.Sprintf("%s.%d", defaultVMRootfsPath, vmID)
+	err = ioutil.WriteFile(rootfsPath, rootfsBytes, 0600)
+	require.NoError(t, err, "failed to copy vm rootfs to %s", rootfsPath)
+
+	tapName := fmt.Sprintf("tap%d", vmID)
```

Comment:
It doesn't look like the tap device is being provided to the VM (which is fine since it doesn't need network connectivity for this test), so we can skip creating it.

Reply:
Will do.

Reply:
Addressed